src/draw/nema_gfx/lv_draw_nema_gfx_blend.c

```diff
@@ -0,0 +1,260 @@
+/**
+ * MIT License
+ *
+ * -----------------------------------------------------------------------------
+ * Copyright (c) 2008-24 Think Silicon Single Member PC
+ * -----------------------------------------------------------------------------
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next paragraph)
+ * shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ * @file lv_draw_nema_gfx_blend.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../misc/lv_area_private.h"
+#include "../sw/blend/lv_draw_sw_blend_private.h"
+#include "../lv_draw_private.h"
+#include "../sw/lv_draw_sw.h"
+#include "../../core/lv_global.h"
+
+#if LV_USE_NEMA_GFX
+
+#include "lv_draw_nema_gfx.h"
+
+#if LV_DRAW_BLEND_CUSTOM == LV_DRAW_BLEND_CUSTOM_NEMA_GFX
+
+#define _draw_info LV_GLOBAL_DEFAULT()->draw_info
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_draw_sw_blend(lv_draw_unit_t * draw_unit, const lv_draw_sw_blend_dsc_t * blend_dsc)
+{
+    /*Do not draw transparent things*/
+    if(blend_dsc->opa <= LV_OPA_MIN) return;
+    if(blend_dsc->mask_buf && blend_dsc->mask_res == LV_DRAW_SW_MASK_RES_TRANSP) return;
+
+    lv_area_t blend_area;
+    if(!lv_area_intersect(&blend_area, blend_dsc->blend_area, draw_unit->clip_area)) return;
+
```

Comment:
I like this initial checking, but I'd like to see an extra information to aid people to track when the things go wrong here, simply putting log-err or log-warning would be helpful here.